diff/parse.go

```diff
@@ -579,6 +593,16 @@ func linePrefix(c byte) bool {
 	return false
 }
 
+// nextLineHasPrefix peeks into the given reader to check whether the next
+// bytes match the given prefix.
+func nextLineHasPrefix(reader *bufio.Reader, prefix []byte) (bool, error) {
```

Comment:
this function is called `nextLineHasPrefix` but it seems to only do a peek, no line parsing/etc. It seems it being `nextLine` has more to do with the state of the reader at the time the function is called. I suggest a different name? Or maybe we can somewhat restructure this code to instead have an abstraction which just reads line by line and has a peekline function?

Reply:
> I suggest a different name?

Good point, yeah. I'll do that.

> Or maybe we can somewhat restructure this code to instead have an abstraction which just reads line by line and has a peekline function?

I thought about that, but the way the rest of the code is structured doesn't make it easy. There's a ton of implicit caching and state involved that I don't want to touch (at least not in this PR). The best solution, IMHO, would be to wrap the `bufio.Reader` that's used here in a custom reader/scanner that stores current and peek lines.